python/GafferUI/PathListingWidget.py

```diff
@@ -1036,3 +1191,16 @@ def __getColumn( self, index ) :
 			raise IndexError( index )
 
 		return columns[index]
+
+	def __getColumnResizeMode( self, index ) :
+
+		columns = _GafferUI._pathListingWidgetGetColumns( GafferUI._qtAddress( self ) )
+
+		if index >= len( columns ) :
+			raise IndexError( index )
+
+		if index == len( columns ) - 1 and self.getStretchLastColumn() :
+			# override the last column to stretch
+			return GafferUI.PathColumn.ResizeMode.Stretch
+
+		return columns[index].getResizeMode()
```

Comment:
Probably not important, but `__getColumn()` and `__getColumnResizeMode()` lead towards accidentaly `O( n^2 )` because they both retrieve all `N` columns and will likely be called `N` times. Probably the number of columns is small enough that it doesn't matter, but I do wonder what some of the code might look like if it retrieved the columns once and then looped over them.

Reply:
Yeah, given the relatively low number of columns we're anticipating it's probably not the end of the world, but I took another look at this and reworked things in 419c73.